src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/AgreedExtensionDateApplicantNotificationHandler.java

```diff
@@ -49,22 +58,19 @@
 
     @Override
     public String camundaActivityId(CallbackParams callbackParams) {
-        return isCcNotification(callbackParams) ? TASK_ID_CC : TASK_ID;
+        return EVENT_TASK_ID_MAP.get(CaseEvent.valueOf(callbackParams.getRequest().getEventId()));
     }
 
     @Override
     public List<CaseEvent> handledEvents() {
-        return EVENTS;
+        return new ArrayList<>(EVENT_TASK_ID_MAP.keySet());
     }
 
     private CallbackResponse notifyApplicantSolicitorForAgreedExtensionDate(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
-        var recipient = isCcNotification(callbackParams)
-            ? getRespondentSolicitorEmailAddress(caseData)
-            : caseData.getApplicantSolicitor1UserDetails().getEmail();
 
         notificationService.sendMail(
-            recipient,
+            getRespondentSolicitorEmailAddress(callbackParams),
```

Comment:
Sorry, why have we removed the logic to determine the recipient?

Reply:
All logic for selecting the recipient has been combined in getRespondentSolicitorEmailAddress()

The getRespondentSolicitorEmailAddress has been refactored to now choose the recipient based on the event rather than on the case data, since we now notify all parties, we have a separate event for each solicitor.

I did forget to update the method name. I will be changing it to getSolicitorEmailAddress()